backend/dataall/base/utils/iam_cdk_utils.py

```diff
@@ -0,0 +1,71 @@
+from typing import Dict, Any, List
+from aws_cdk import aws_iam as iam
+
+from dataall.base.utils.iam_policy_utils import (
+    split_policy_statements_in_chunks,
+    split_policy_with_resources_in_statements,
+    split_policy_with_mutiple_value_condition_in_statements,
+)
+
+
+def convert_from_json_to_iam_policy_statement_with_conditions(iam_policy: Dict[Any, Any]):
+    return iam.PolicyStatement(
+        sid=iam_policy.get('Sid'),
+        effect=iam.Effect.ALLOW if iam_policy.get('Effect').casefold() == 'Allow'.casefold() else iam.Effect.DENY,
+        actions=_convert_to_array(str, iam_policy.get('Action')),
+        resources=_convert_to_array(str, iam_policy.get('Resource')),
+        conditions=iam_policy.get('Condition'),
+    )
+
+
+def convert_from_json_to_iam_policy_statement(iam_policy: Dict[Any, Any]):
+    return iam.PolicyStatement(
+        sid=iam_policy.get('Sid'),
+        effect=iam.Effect.ALLOW if iam_policy.get('Effect').casefold() == 'Allow'.casefold() else iam.Effect.DENY,
+        actions=_convert_to_array(str, iam_policy.get('Action')),
+        resources=_convert_to_array(str, iam_policy.get('Resource')),
+    )
+
+
+def process_and_split_statements_in_chunks(statements: List[Dict]):
+    statement_chunks_json: List[List[Dict]] = split_policy_statements_in_chunks(statements)
+    statements_chunks: List[List[iam.PolicyStatement]] = []
+    for statement_js_chunk in statement_chunks_json:
+        statements: List[iam.PolicyStatement] = []
+        for statement in statement_js_chunk:
+            if statement.get('Condition', None):
+                statements.append(convert_from_json_to_iam_policy_statement_with_conditions(statement))
+            else:
+                statements.append(convert_from_json_to_iam_policy_statement(statement))
+        statements_chunks.append(statements)
+    return statements_chunks
+
+
```

Comment:
`process_and_split_policy_with_resources_in_statements` is a function doing 2 separate things at the same time: 1) policies with resources and 2) policies with conditions. Let's make 2 functions each doing one single thing:
`process_and_split_policy_with_resources_in_statements` and `process_and_split_policy_with_conditions_in_statements`

Reply:
updated the PR and separated this. Thanks for pointing this out.